ir/src/main/scala/org/scalajs/core/ir/Types.scala

```diff
@@ -84,13 +84,14 @@ object Types {
   case object NullType extends Type
 
   /** Reference types (allowed for classOf[], is/asInstanceOf[]). */
-  sealed abstract class ReferenceType extends Type
+  sealed trait ReferenceType
```

Comment:
Ok, I see. But why then don't you mark the cast in `asInstanceOf` as suspicious? IIUC we're doing the following wrong:

```scala
AsInstanceOf(???, ClassType("scala.Int")).tpe // -> ClassType("scala.Int")
```

But it should be `IntType`.